client/my-sites/patterns/use-pattern-library.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import type { Pattern } from 'calypso/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/types';
+
+export function usePatternLibrary(
+	locale: string,
+	queryOptions: Omit< UseQueryOptions< any, unknown, Pattern[] >, 'queryKey' > = {}
+) {
+	return useQuery< any, unknown, Pattern[] >( {
+		queryKey: [ 'patterns', 'library', locale ],
+		queryFn: () => {
+			return wpcomRequest( {
+				path: `/ptk/patterns/${ locale }`,
+				method: 'GET',
+				apiVersion: '1.1',
+				query: new URLSearchParams( {
+					categories: 'intro',
+					per_page: '4',
+					post_type: 'wp_block',
+				} ).toString(),
+			} );
+		},
```

Comment:
Per our discussion in Slack, p1708591527394989/1708581605.350569-slack-C06ELHR6L9J, I think we should try to extract this function into a separately exported function. The goal is to use the exact same fetching logic in `fetchPatterns` as we do here